GameDataParser/Program.cs

```diff
@@ -3,87 +3,51 @@
 using System.Globalization;
 using System.IO;
 using System.IO.MemoryMappedFiles;
-using System.Threading;
+using System.Linq;
+using System.Threading.Tasks;
 using System.Xml;
 using GameDataParser.Crypto;
-using GameDataParser.Crypto.Common;
 using GameDataParser.Crypto.Stream;
 using GameDataParser.Files;
-using GameDataParser.Files.Export;
+using GameDataParser.Parsers;
 
 namespace GameDataParser
 {
     internal static class Program
     {
-        private static void Main()
+        static async Task Main()
         {
-            // Force Globalization to en-US because we use periods instead of commas for decimals
-            CultureInfo.CurrentCulture = new CultureInfo("en-US");
-
             // Create Resources folders if they don't exist
             Directory.CreateDirectory(Paths.INPUT);
             Directory.CreateDirectory(Paths.OUTPUT);
 
-            // Resources
-            try
-            {
-                string xmlPath = $"{Paths.INPUT}/Xml.m2d";
-                string exportedPath = $"{Paths.INPUT}/Exported.m2d";
-                string xmlHeaderPath = xmlPath.Replace("m2d", "m2h");
-                string exportedHeaderPath = exportedPath.Replace("m2d", "m2h");
-
-                List<PackFileEntry> xmlFiles = FileList.ReadFile(File.OpenRead(xmlHeaderPath));
-                List<PackFileEntry> exportedFiles = FileList.ReadFile(File.OpenRead(exportedHeaderPath));
-
-                MemoryMappedFile xmlMemFile = MemoryMappedFile.CreateFromFile(xmlPath);
-                MemoryMappedFile exportedMemFile = MemoryMappedFile.CreateFromFile(exportedPath);
+            Spinner spinner = new Spinner();
+            spinner.Start();
 
-                // Threads
-                Thread itemThread = new Thread(() => ItemMetadataExport.Export(xmlFiles, xmlMemFile));
-                Thread mapEntityThread = new Thread(() => MapMetadataExport.Export(exportedFiles, exportedMemFile));
-                Thread skillThread = new Thread(() => SkillMetadataExport.Export(xmlFiles, xmlMemFile));
-                Thread insigniaThread = new Thread(() => InsigniaMetadataExport.Export(xmlFiles, xmlMemFile));
-                Thread prestigeThread = new Thread(() => PrestigeMetadataExport.Export(xmlFiles, xmlMemFile));
-                Thread expThread = new Thread(() => ExpMetadataExport.Export(xmlFiles, xmlMemFile));
-                Thread questThread = new Thread(() => QuestMetadataExport.Export(xmlFiles, xmlMemFile));
-                Thread scriptThread = new Thread(() => ScriptMetadataExport.Export(xmlFiles, xmlMemFile));
-                Thread guildThread = new Thread(() => GuildMetadataExport.Export(xmlFiles, xmlMemFile));
+            CultureInfo.CurrentCulture = new CultureInfo("en-US");
```

Comment:
Is there a reason this was moved further down in the code?

Reply:
Not really

Reply:
Ok then generally it should be at the top since we're setting a value that is used across the program

Reply:
Updated